Xamarin.Forms.Platform.iOS/EventTracker.cs

```diff
@@ -133,22 +138,81 @@ protected virtual NativeGestureRecognizer GetNativeRecognizer(IGestureRecognizer
 				return uiRecognizer;
 			}
 #else
-            if (tapRecognizer != null)
-            {
-                var returnAction = new Action(() =>
-                {
-                    var tapGestureRecognizer = weakRecognizer.Target as TapGestureRecognizer;
-                    var eventTracker = weakEventTracker.Target as EventTracker;
-                    var view = eventTracker?._renderer?.Element as View;
-
-                    if (tapGestureRecognizer != null && view != null)
-                        tapGestureRecognizer.SendTapped(view);
-                });
-                var uiRecognizer = CreateTapRecognizer(tapRecognizer.NumberOfTapsRequired, returnAction);
-                return uiRecognizer;
-            }                
+			if (tapRecognizer != null)
+			{
+				var returnAction = new Action<UITapGestureRecognizer>((sender) =>
+				{
+					var eventTracker = weakEventTracker.Target as EventTracker;
+					var view = eventTracker?._renderer?.Element as View;
+
+					var originPoint = sender.LocationInView(null);
+					originPoint = UIApplication.SharedApplication.KeyWindow.ConvertPointToView(originPoint, eventTracker._renderer.NativeView);
+
+					var childGestures = view.GetChildElements(new Point(originPoint.X, originPoint.Y));
+
+					if (childGestures?.GetChildGesturesFor<TapGestureRecognizer>(x => x.NumberOfTapsRequired == (int)sender.NumberOfTapsRequired).Count() > 0)
+						return;
+
+					if (weakRecognizer.Target is TapGestureRecognizer tapGestureRecognizer && view != null)
+						tapGestureRecognizer.SendTapped(view);
+
+				});
+
+				var uiRecognizer = CreateTapRecognizer(tapRecognizer.NumberOfTapsRequired, returnAction);
+				return uiRecognizer;
+			}
 #endif
 
+			if (recognizer is ChildGestureRecognizer childRecognizer)
+			{
+#if !__MOBILE__
```

Comment:
Again, not a blocker, but it'd be nice to have this split into separate methods to make it easier to read and maintain.

Reply:
Done